src/azure-cli/azure/cli/command_modules/cdn/_params.py

```diff
@@ -154,3 +140,55 @@ def load_arguments(self, _):
     # Origin #
     with self.argument_context('cdn origin') as c:
         c.argument('origin_name', name_arg_type, id_part='name')
+
+    # WAF #
+    with self.argument_context('cdn waf') as c:
+        c.argument('rule_set_type', help='The type of the rule set.')
+        c.argument('rule_set_version', help='The version of the rule set.')
+
+    with self.argument_context('cdn waf managed-rule-set rule-group') as c:
+        c.argument('name', name_arg_type, id_part='name', help='The name of the rule group.')
+
+    with self.argument_context('cdn waf policy set') as c:
+        c.argument('disabled', arg_type=get_three_state_flag())
+        c.argument('block_response_status_code', type=int)
+        c.argument('name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy show') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy delete') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+
+    with self.argument_context('cdn waf policy managed-rule-set') as c:
+        c.argument('policy_name', help='Name of the CDN WAF policy.')
+        c.argument('rule_set_type', help='Type name of the managed rule set.')
+        c.argument('rule_set_version', help='Version the managed rule set.')
+    with self.argument_context('cdn waf policy managed-rule-set add') as c:
+        c.argument('enabled', arg_type=get_three_state_flag())
+
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override') as c:
+        c.argument('name', name_arg_type, id_part='name', help='The name of the rule group override.')
+        c.argument('policy_name', help='Name of the CDN WAF policy.')
+        c.argument('rule_set_type', help='Type name of the managed rule set.')
+        c.argument('rule_set_version', help='Version of the managed rule set.')
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override set') as c:
+        c.argument('rule_overrides',
+                   options_list=['-r', '--rule-override'],
+                   action=ManagedRuleOverrideAction,
+                   nargs='+')
+
+    with self.argument_context('cdn waf policy custom-rule') as c:
```

Comment:
What is the difference between custom-rule and rate-limit-rule? Why not make the one subgroup named "rule"?

Reply:
rate-limit-rule and custom-rule have the same format except that rate-limit-rule has two extra required fields and have different behavior for determining when a rule's action is applied. There is also a third type of rule, managed rules, which are grouped as a managed-rule-set. The reasons why I didn't move these three types of rules under one subgroup are:

1. I thought it would be confusing if managed rules didn't fall under the "rule" subgroup as well, but because they are defined under rule sets it seems odd to have the rule above the rule set. If you're ok with managed rules not falling under the "rule" subgroup then this concern doesn't apply.

2. Our service and documentation refer to the titles of these properties as "Managed Rules" "Custom Rules" and "Rate Limit Rules". We don't typically refer to all of them together as just "rules".

If you still think it would be better to move two of the three types of rules under a "rule" subgroup so the commands are `az cdn waf policy rule custom <action>`, `az cdn waf policy rule rate-limit <action>`, and `az cdn waf policy managed-rule-set rule-group set --rule <rule definition>`, I'm happy to make the change.